src/scripts/main.js

```diff
@@ -1,3 +1,222 @@
 'use strict';
 
-// write code here
+const body = document.querySelector('body');
+
+let table, headers, tbody, rows;
+
+const init = () => {
+  table = document.querySelector('table');
+  headers = [...table.querySelectorAll('th')];
+  tbody = table.querySelector('tbody');
+  rows = [...tbody.querySelectorAll('tr')];
+};
+
+init();
+
+const currencyFormat = (number) =>
+  '$' + new Intl.NumberFormat('en-US').format(number);
+
+// Add active class to row when clicked
+
+rows.forEach((row) => {
+  row.addEventListener('click', () => {
+    rows.forEach((r) => r.classList.remove('active'));
+    row.classList.toggle('active');
+  });
+});
+
+let ASC = true;
+let currentHeader;
+
+// Sort table by column
+headers.forEach((header, i) => {
+  header.addEventListener('click', (e) => {
+    ASC = !ASC;
+
+    if (currentHeader !== i) {
+      ASC = true;
+    }
+    currentHeader = i;
+
+    rows.sort((a, b) => {
+      const cellA = a.querySelectorAll('td')[i].textContent;
+      const cellB = b.querySelectorAll('td')[i].textContent;
+
+      const checkNumberCellA = numberFormat(cellA);
+      const checkNumberCellB = numberFormat(cellB);
+
+      if (Number(checkNumberCellA)) {
+        return ASC
+          ? checkNumberCellA - checkNumberCellB
+          : checkNumberCellB - checkNumberCellA;
+      }
+
+      return ASC ? cellA.localeCompare(cellB) : cellB.localeCompare(cellA);
+    });
+
+    rows.forEach((row) => tbody.appendChild(row));
+  });
+});
```

Comment:
This block of code seems to be doing multiple things. It might be a good idea to break it down into smaller functions, each doing one specific thing. This would make your code more readable and easier to maintain.